Building 3.5-stable with Visual Studio 2017 fails

### Godot version

3.5-stable

### System information

Windows 10

### Issue description

Godot 3.5-stable compile attempt in Visual Studio 2017 command line tools window fails with the following output:

> main\tests\test_theme.cpp(58): error C2232: '->Variant::~Variant': left operand has 'class' type, use '.'
> main\tests\test_theme.cpp(58): note: This diagnostic occurred in the compiler generated function 'TestTheme::Fixture::Fixture(void)

It is found experimentally that this error is caused by lines 55, 56, and 57 of main/tests/test_theme.cpp. If these lines are commented, compilation succeeds:

> 	} const valid_data[Theme::DATA_TYPE_MAX] = {
> 		{ Theme::DATA_TYPE_COLOR, Color() },
> 		{ Theme::DATA_TYPE_CONSTANT, 42 },
> 		//{ Theme::DATA_TYPE_FONT, Ref<Font>(memnew(BitmapFont)) },
> 		//{ Theme::DATA_TYPE_ICON, Ref<Texture>(memnew(ImageTexture)) },
> 		//{ Theme::DATA_TYPE_STYLEBOX, Ref<StyleBox>(memnew(StyleBoxFlat)) },
> 	};

If either of the three lines is uncommented, the error is back.






### Steps to reproduce

Just run 

> scons p=windows tools=yes

in Visual Studio 2017 command line tools terminal window.


### Minimal reproduction project

_No response_

--- a/main/tests/test_theme.cpp
+++ b/main/tests/test_theme.cpp
@@ -49,16 +49,24 @@ public:
 	struct DataEntry {
 		Theme::DataType type;
 		Variant value;
-	} const valid_data[Theme::DATA_TYPE_MAX] = {
+	} valid_data[Theme::DATA_TYPE_MAX] = {
 		{ Theme::DATA_TYPE_COLOR, Color() },
 		{ Theme::DATA_TYPE_CONSTANT, 42 },
-		{ Theme::DATA_TYPE_FONT, Ref<Font>(memnew(BitmapFont)) },
-		{ Theme::DATA_TYPE_ICON, Ref<Texture>(memnew(ImageTexture)) },
-		{ Theme::DATA_TYPE_STYLEBOX, Ref<StyleBox>(memnew(StyleBoxFlat)) },
+		{ Theme::DATA_TYPE_FONT, Variant() },
+		{ Theme::DATA_TYPE_ICON, Variant() },
+		{ Theme::DATA_TYPE_STYLEBOX, Variant() },
 	};
 
 	const StringName valid_item_name = "valid_item_name";
 	const StringName valid_type_name = "ValidTypeName";
+
+	// Part of `valid_data` initialization is moved here so that it compiles in Visual Studio 2017.
+	// See issue #63975.
+	Fixture() {
+		valid_data[Theme::DATA_TYPE_FONT].value = Ref<Font>(memnew(BitmapFont));
+		valid_data[Theme::DATA_TYPE_ICON].value = Ref<Texture>(memnew(ImageTexture));
+		valid_data[Theme::DATA_TYPE_STYLEBOX].value = Ref<StyleBox>(memnew(StyleBoxFlat));
+	}
 };
 
 bool test_good_theme_type_names() {

